pkg/pillar/cmd/nim/nim.go

```diff
@@ -671,6 +671,14 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject) in
 					handleLinkChange(&nimCtx)
 					handleInterfaceChange(&nimCtx, ifindex,
 						"LinkChange", true)
+					ifname := change.Attrs().Name
+					if ifname != "" {
+						portStatus := nimCtx.deviceNetworkContext.DevicePortConfigList.PortConfigList[0].GetPortByIfName(ifname)
```

Comment:
Note that PortConfigList can be zero length at least when we start.  Hence referencing [0] is dangerous without a check for the length.

Reply:
Maybe introducing a function isIfNameInIndex0(change.Attrs().Name) ?

Question: if we are using index 1 and an interface used in 1 changes, should we also restart? If so the check should be isIfNameInCurrent()

Reply:
I think we should check if ifname is either part of DPC at current index or is part of DPC at index 0.